src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -507,13 +506,7 @@ private void initializeConnections()
    {
       if (config.isInitializationFailFast()) {
          try {
-            Connection connection = getConnection();
-            if (config.getMinimumIdle() == 0) {
-               evictConnection(connection);
-            }
-            else {
-               connection.close();
-            }
+            newConnection().close();
```

Comment:
2.4.1 and before it was returning immediately with correct exception if any.
now It is not returning until 'connectionTimeout'